native/src/routes/Pois.tsx

```diff
@@ -58,19 +62,30 @@ type PoisProps = {
 
 const RESTORE_TIMEOUT = 100
 
-const Pois = ({ pois, language, cityModel, route, navigation }: PoisProps): ReactElement => {
+const Pois = ({ pois: allPois, language, cityModel, route, navigation }: PoisProps): ReactElement => {
+  const [poiCategoryFilter, setPoiCategoryFilter] = useState<PoiCategoryModel | null>(null)
+  const [poiCurrentlyOpenFilter, setPoiCurrentlyOpenFilter] = useState(false)
+  const [showFilterSelection, setShowFilterSelection] = useState(false)
   const { coordinates, requestAndDetermineLocation } = useUserLocation(true)
   const { slug } = route.params
   const [sheetSnapPointIndex, setSheetSnapPointIndex] = useState<number>(1)
   const [followUserLocation, setFollowUserLocation] = useState<boolean>(false)
   const [listScrollPosition, setListScrollPosition] = useState<number>(0)
-  const features = prepareFeatureLocations(pois, coordinates)
-  const selectedFeature = slug ? features.find(it => it.properties.slug === slug) : null
-  const poi = pois.find(it => it.slug === slug)
   const { t } = useTranslation('pois')
   const theme = useTheme()
   const scrollRef = useRef<BottomSheetScrollViewMethods>(null)
 
+  const pois = useMemo(
+    () =>
+      allPois
+        .filter(poi => !poiCategoryFilter || poi.category === poiCategoryFilter)
+        .filter(poi => !poiCurrentlyOpenFilter || poi.isCurrentlyOpen),
+    [allPois, poiCategoryFilter, poiCurrentlyOpenFilter]
+  )
+  const poi = pois.find(it => it.slug === slug)
```

Comment:
Not sure how to handle filtering while a poi is selected. Perhaps we should just unselect the poi?

Reply:
Done, good suggestion